openfe/setup/protein_molecule.py

```diff
@@ -0,0 +1,59 @@
+# This code is part of OpenFE and is licensed under the MIT license.
+# For details, see https://github.com/OpenFreeEnergy/openfe
+from rdkit import Chem
+
+from openfe.utils.typing import RDKitMol, OEMol
+
+
+class ProteinMolecule:
+    """Wrapper around a Protein representation
+
+    This representation is immutable.  If you want to make any modifications,
+    do this in an appropriate toolkit then remake this class.
+    """
+    def __init__(self, rdkit: RDKitMol, name=""):
+        self._rdkit = rdkit
+        self._name = name
```

Comment:
As a "naive user question", would there be value in defaulting name to the actual PDB file name minus the extension here? I.e. that's a generally common strategy folks have for naming their PDBs in other workflows.

Reply:
@IAlibay : I think this might make sense in the `from_pdbfile` method -- otherwise, do we even know where to look for this filename?

Reply:
Yeah nvm, I thought it was a stored property in the rdmol, but turns out it's not.